### ISSUE TITLE
Why multi object pose estimation is not possible with linemod dataset?

### ISSUE THREAD

**Issue by danikhani** [ybkscht/EfficientPose#7]
Hello,

thank you very much for your nicely commented code!
In your paper under section 4.1.1 we have being told that multi label pose estimation is not possible with linemod dataset. Therefore Occlusion is used for that porpuse. 
I have downloaded both datasets and cant really understand why this is not possible with linemod. Could you please assist me with this question? 
It will be much helpful to let me train my own dataset with your project for multi lable pose estimation.

Im really grateful for your time

**Comment by ybkscht** [comment]
Hi danikhani,

I already answered your question [here](https://github.com/ybkscht/EfficientPose/issues/3#issuecomment-762300332).
To give you a better understanding, I took the following images from the Linemod dataset including their 6D annotation.

![cat](https://user-images.githubusercontent.com/74176077/105195945-6a53b180-5b3b-11eb-9687-6f0a42af1e57.jpg)
![duck](https://user-images.githubusercontent.com/74176077/105195982-72135600-5b3b-11eb-9ad8-167988fb6149.jpg)
![driller](https://user-images.githubusercontent.com/74176077/105195972-6f186580-5b3b-11eb-8967-94b3fa127f2a.jpg)

You can see that in every image only a single object is annotated in the Linemod dataset but the other objects are also visible in the images.
If you want to train a single model for all objects the problem is that if you feed e.g. the annotated cat image from above to your model, you can only provide the annotations of the cat but not for the ape, duck, etc.
So the ground truth for all the other objects without annotations is just "background". This will unnecessarily penalize your model a lot during training when actually correctly detecting the other objects.

The Occlusion dataset (subset of Linemod) instead provides annotations for all objects in the image and is therefore better suited for multi 6d pose estimation.

Furthermore, I don't think it is impossible to train a single model for all Linemod objects because you can try using synthetic rendered data and the cut&paste strategy like PVNet but you probably can't use the vanilla Linemod images which could worsen the performance of your model on the test dataset.
But I didn't do any experiments here so this is just a guess.

Sincerely,
Yannick

**Comment by greatwallet** [comment]
Hi!

Sorry for re-mentioning the issue, but I still have a question here, for settings upon multi-object pose estimation.

One of your competitor, DPOD (`Zakharov, Sergey, Ivan Shugurov, and Slobodan Ilic. "Dpod: 6d pose object detector and refiner." Proceedings of the IEEE/CVF International Conference on Computer Vision. 2019.`), is also a holistic method (if its refiner is not used) for multi-object pose estimation. As is stated in its paper, DPOD has conducted corresponding experiments tested on Occlusion dataset, and clearly all objects are trained in one single model. So why don't you conduct this experiment in the same setting?

![image](https://user-images.githubusercontent.com/30019142/129829958-267c73e5-c211-4baf-9e9c-cad892b1a487.png)

P.S. One thing I slightly suspect about this experiment of DPOD is that the training dataset might not include real dataset from LineMOD. However, if it were trained with synthetic images only, this crutial settings should have been pointed out in the paper. Also, the synthetic dataset generated by DPOD is of really low quality, I highly doubt that if they can achieve 47.25% ADD on Occlusion if trained only with synthetic images.
